File: iOS/MoviePlayer/MovieLibraryApp/MovieLibraryApp/MovieDescription.swift
// Copyright 2016 Supraj Kolluri,
//
//
// The contents of the file can only be used for the purpose of grading and reviewing.
// The instructor and the University have the right to build and evaluate the
// software package for the purpose of determining the grade and program assessment.
//
//
// @author Supraj Kolluri mailto:supraj.kolluri@asu.edu
//         Software Engineering, CIDSE, IAFSE, ASU Poly
// @version Apr 27, 2016
 
 
 import Foundation
 class MovieDescription {
    var titleVar: String
    var yearVar: String
    var ratedVar: String
    var releasedVar: String
    var runtimeVar: String
    var genreVar: String
    var actorsVar: String
    var plotVar: String
    var posterVar:String
    var filenameVar:String
    
    init (jsonStr: String){
        self.titleVar = ""
        self.yearVar=""
        self.ratedVar = ""
        self.releasedVar = ""
        self.runtimeVar = ""
        self.genreVar = ""
        self.actorsVar = ""
        self.plotVar = ""
        self.posterVar = ""
        self.filenameVar = ""
        
        
        if let data: NSData = jsonStr.dataUsingEncoding(NSUTF8StringEncoding){
            do{
                let dict = try NSJSONSerialization.JSONObjectWithData(data,options:.MutableContainers) as?[String:AnyObject]
                self.titleVar = (dict!["Title"] as? String)!
                self.yearVar = (dict!["Year"] as? String)!
                self.ratedVar = (dict!["Rated"] as? String)!
                self.releasedVar = (dict!["Released"] as? String)!
                self.runtimeVar = (dict!["Runtime"] as? String)!
                self.genreVar = (dict!["Genre"] as? String)!
                self.actorsVar = (dict!["Actors"] as? String)!
                self.plotVar = (dict!["Plot"] as? String)!
                self.posterVar = (dict!["Poster"] as? String)!
                self.filenameVar = (dict!["Filename"] as? String)!
            } catch {
                print("unable to convert to dictionary")
                
            }
        }
    }
    
    func toJsonString() -> String {
        var jsonStr = "";
        let dict = ["Title": titleVar, "Year": yearVar, "Rated":ratedVar, "Released":releasedVar, "Runtime":runtimeVar, "Genre":genreVar , "Actors":actorsVar, "Plot":plotVar]
        do {
            let jsonData = try NSJSONSerialization.dataWithJSONObject(dict, options: NSJSONWritingOptions.PrettyPrinted)
            // here "jsonData" is the dictionary encoded in JSON data
            jsonStr = NSString(data: jsonData, encoding: NSUTF8StringEncoding)! as String
        } catch let error as NSError {
            print(error)
        }
        return jsonStr
    }
 }

File: Android/JSONRPCMovieCollection/JSONRPCServer/MovieRPCServer/src/MovieDescription.java
package edu.asu.msse.skollur1.server;

import org.json.JSONObject;
/*
 * Copyright 2016 Supraj Kolluri,
 *
 *
 * The contents of the file can only be used for the purpose of grading and reviewing.
 * The instructor and the University have the right to build and evaluate the
 * software package for the purpose of determining the grade and program assessment.
 *
 *
 * @author Supraj Kolluri mailto:supraj.kolluri@asu.edu
 *         Software Engineering, CIDSE, IAFSE, ASU Poly
 * @version March 15, 2016
 */
public class MovieDescription {
	public String title, rated, released, runtime, plot, genre, actors, year;

	public MovieDescription() {

	}

	public MovieDescription(JSONObject jsonObj) {
		try {
			System.out.println("constructor from json received: " + jsonObj.toString());
			title = jsonObj.getString("Title");
			rated = jsonObj.getString("Rated");
			released = jsonObj.getString("Released");
			runtime = jsonObj.getString("Runtime");
			plot = jsonObj.getString("Plot");
			genre = jsonObj.getString("Genre");
			actors = jsonObj.getString("Actors");
			year = jsonObj.getString("Year");

		} catch (Exception ex) {
			System.out.println(this.getClass().getSimpleName() + ": error converting from json string");
		}
	}

	public MovieDescription(String json) {
		try {
			JSONObject jo = new JSONObject(json);
			title = jo.getString("Title");
			rated = jo.getString("Rated");
			released = jo.getString("Released");
			runtime = jo.getString("Runtime");
			plot = jo.getString("Plot");
			year = jo.getString("Year");
			genre = jo.getString("Genre");
			actors = jo.getString("Actors");

		} catch (Exception ex) {
			ex.printStackTrace();
		}
	}

	public JSONObject toJson() {
		JSONObject jo = new JSONObject();
		try {
			jo.put("Title", title);
			jo.put("Rated", rated);
			jo.put("Released", released);
			jo.put("Runtime", runtime);
			jo.put("Plot", plot);
			jo.put("Year", year);
			jo.put("Genre", genre);
			jo.put("Actors", actors);

		} catch (Exception ex) {
			ex.printStackTrace();
		}

		return jo;
	}

	public String toJsonString() {
		String ret = "";
		try {

			ret = this.toJson().toString();
		} catch (Exception ex) {
			ex.printStackTrace();
		}
		return ret;
	}

	@Override
	public String toString() {
		return "MovieDescription [title=" + title + ", rated=" + rated + ", released=" + released + ", runtime="
				+ runtime + ", plot=" + plot + ", year=" + year + ", genre=" + genre + ", actors=" + actors + "]";
	}

}
